kadi/events/models.py

```diff
@@ -1466,23 +1481,33 @@ def get_target_attitudes(cls, event, msidset):
 
     @classmethod
     @import_ska
-    def get_one_shot(cls, guide_start, one_shot_msid):
+    def get_one_shot(cls, guide_start, aux_msid):
         """
         Define one_shot attribute (one-shot attitude update following maneuver)
         """
-        # No acq => guide transition so no one-shot defined.  Use 0.0 as a convenience.
+        # No acq => guide transition so no one-shot defined.  Use -99.0 as a convenience.
         if guide_start is None:
             return -99.0
 
-        t0 = DateTime(guide_start).secs
-        n_times = len(one_shot_msid.times)
-        i0, i1 = np.searchsorted(one_shot_msid.times, [t0 - 10, t0 + 10])
-        if i0 == 0 or i1 == 0 or i0 == n_times or i1 == n_times:
-            raise ValueError('Unable to find complete data for one-shot attitude value'
-                             ' {} {} {} {} {}'
-                             .format(i0, i1, t0, np.min(one_shot_msid.times),
-                                     np.max(one_shot_msid.times)))
-        return np.max(one_shot_msid.vals[i0:i1])
+        aoatter1 = aux_msid['aoatter1']
+        aoatter2 = aux_msid['aoatter2']
+        aoatter3 = aux_msid['aoatter3']
+        if len(aoatter2.times) == 0:
+            raise ValueError("No AOATTER2 times for guide transition at {} for one shot".format(
+                    guide_start))
+        dtime = aoatter2.times - DateTime(guide_start).secs
+        # Define the index of the one shot as the sample closes to the guide transition
+        # That guide_start from AOACASEQ should be coming from the same telemetry as AOATTER*
+        # so the closest timestamp should really be from the same timestamp
```

Comment:
Since the `guide_start` telemetry has not been interpolated to the same time stamps as the aux msids, you can't make this assumption.  Instead just look for the first value with `guide_start <= time < guide_start + 30` (independently for each axis).